lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1805,6 +1818,7 @@ private boolean isFatalError(ErrorInfo err) {
     private CMConnectivityListener connectivityListener;
     private long connectionStateTtl = Defaults.connectionStateTtl;
     long maxIdleInterval = Defaults.maxIdleInterval;
+    private int disconnectedRetryCount = 0;
```

Comment:
I'm curious as to why this field has a `disconnected` prefix on its name when the (equivalent?) field you've added in `ChannelBase` is just `retryCount`. Does it carry extra meaning in this class?

Reply:
@QuintinWillison what should we do here?

Reply:
The reason for the difference here is that a connection will retry while in the `DISCONNECTED` state and also while in the `SUSPENDED` state, so the `disconnectedRetryTimeout` is named as such to make it clear that this variable only counts the number of retries while in the `DISCONNECTED` state. Realtime channels only retry while they are `SUSPENDED` so there is no distinction to be made in that case. That being said, I have no objection to calling it `ChannelBase.suspendedRetryCount` if that makes it clearer.

Reply:
Thanks, both. If it's matching the JS implementation then let's keep it as is. I'm also working on a new release this afternoon so this could nicely slip into that also. 😁